packages/fetchai/skills/tac_control_contract/behaviours.py

```diff
@@ -0,0 +1,271 @@
+                    self.context.agent_name, parameters.start_time
+                )
+            )
+        elif (
+            game.phase.value == Phase.GAME_REGISTRATION.value
+            and parameters.start_time < now < parameters.end_time
+        ):
+            if game.registration.nb_agents < parameters.min_nb_agents:
+                self._cancel_tac()
+                game.phase = Phase.POST_GAME
+                self._unregister_tac()
+            else:
+                game.phase = Phase.GAME_SETUP
+                self._start_tac()
+                self._unregister_tac()
+                game.phase = Phase.GAME
+        elif game.phase.value == Phase.GAME.value and now > parameters.end_time:
+            self._cancel_tac()
+            game.phase = Phase.POST_GAME
+
+    def teardown(self) -> None:
+        """
+        Implement the task teardown.
+
+        :return: None
+        """
+        if self._registered_desc is not None:
+            self._unregister_tac()
+
+    def _register_tac(self) -> None:
+        """
+        Register on the OEF as a TAC controller agent.
+
+        :return: None.
+        """
+        self._oef_msg_id += 1
+        desc = Description(
+            {"version": self.context.parameters.version_id},
+            data_model=CONTROLLER_DATAMODEL,
+        )
+        self.context.logger.info(
+            "[{}]: Registering TAC data model".format(self.context.agent_name)
+        )
+        oef_msg = OEFMessage(
+            type=OEFMessage.Type.REGISTER_SERVICE,
+            id=self._oef_msg_id,
+            service_description=desc,
+            service_id="",
+        )
+        self.context.outbox.put_message(
+            to=DEFAULT_OEF,
+            sender=self.context.agent_address,
+            protocol_id=OEFMessage.protocol_id,
+            message=OEFSerializer().encode(oef_msg),
+        )
+        self._registered_desc = desc
+
+    def _unregister_tac(self) -> None:
+        """
+        Unregister from the OEF as a TAC controller agent.
+
+        :return: None.
+        """
+        self._oef_msg_id += 1
+        self.context.logger.info(
+            "[{}]: Unregistering TAC data model".format(self.context.agent_name)
+        )
+        oef_msg = OEFMessage(
+            type=OEFMessage.Type.UNREGISTER_SERVICE,
+            id=self._oef_msg_id,
+            service_description=self._registered_desc,
+            service_id="",
+        )
+        self.context.outbox.put_message(
+            to=DEFAULT_OEF,
+            sender=self.context.agent_address,
+            protocol_id=OEFMessage.protocol_id,
+            message=OEFSerializer().encode(oef_msg),
+        )
+        self._registered_desc = None
+
+    def _start_tac(self):
+        """Create a game and send the game configuration to every registered agent."""
+        game = cast(Game, self.context.game)
+        contract = cast(Contract, self.context.contracts.erc1155)
+        game.create(contract)
```

Comment:
Game is a class exposed on the context, so it has access to `self.context.contracts.erc1155`.

Reply:
So no need to pass it in as an argument.